### PR TITLE
feat(autoware_test_utils): add general topic dumper

### PR DESCRIPTION
**Pull request by soblin**
## Description

By running 

```
ros2 launch autoware_test_utils psim_road_shoulder.launch.xml vehicle_model:=<vehicle-model> sensor_model:=<sensor-model>
```

and then stop the simulation whenever you want, using the SimulatedClockPanel, you can save the specified ros topic in the specified format by call the service.

```
ros2 service call /autoware_test_utils/topic_snapshot_saver std_srvs/srv/Empty \{\}
```

https://github.com/user-attachments/assets/c97c2177-a6e0-482b-8c0b-8c23ef91f66d

## Related links

**Parent Issue:**

- https://tier4.atlassian.net/browse/RT1-8338

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: common/autoware_test_utils/config/sample_topic_snapshot.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+format_version: 1
```
Comment (soblin) [inline common/autoware_test_utils/config/sample_topic_snapshot.yaml:1]: need to validate the format in parser side.

**Comment by kyoichi-sugahara** [comment]
run following command in the attached state
```
ros2 service call /autoware_test_utils/topic_snapshot_saver std_srvs/srv/Empty \{\}
```
image

generated file is following
[topic_snapshot.zip](https://github.com/user-attachments/files/17627383/topic_snapshot.zip)

**Comment by kyoichi-sugahara** [comment]
@soblin 
This is a really great feature. Am I correct in understanding that examples of utilizing the YAML files generated on the unit test side and the readme will be added in the future?

File: common/autoware_test_utils/src/topic_snapshot_saver.cpp [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    if (map_path_uri == "none" || config_path_uri == "none") {
+      RCLCPP_ERROR(get_logger(), "map_path_uri and/or config_path_uri are not provided");
+      return;
+    }
+    const auto map_path = autoware::test_utils::resolve_pkg_share_uri(map_path_uri);
+    const auto config_path = autoware::test_utils::resolve_pkg_share_uri(config_path_uri);
+    if (!map_path) {
+      RCLCPP_ERROR(
+        get_logger(),
+        "failed to resolve %s. expected form is package://<package-name>/<resouce-path>",
```
Comment (soblin) [inline common/autoware_test_utils/src/topic_snapshot_saver.cpp:273]:
```suggestion
        "failed to resolve %s. expected form is package://<package-name>/<resource-path>",
```

File: common/autoware_test_utils/src/topic_snapshot_saver.cpp [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+        map_path_uri.c_str());
+    } else if (!config_path) {
+      RCLCPP_ERROR(
+        get_logger(),
+        "failed to resolve %s. expected form is package://<package-name>/<resouce-path>",
```
Comment (soblin) [inline common/autoware_test_utils/src/topic_snapshot_saver.cpp:278]:
```suggestion
        "failed to resolve %s. expected form is package://<package-name>/<resource-path>",
```

**Comment by soblin** [comment]
Please ignore the spellcheck because they stem from other packages

**Review by kyoichi-sugahara — APPROVED** [review]
Thank you! LGTM!

**Pull request merged**